src/PABLO/Octant.cpp

```diff
@@ -1584,6 +1584,205 @@ Octant Octant::computePeriodicOctant(uint8_t iface) const {
 		return degOct;
 	}
 
+};
+
+/** Build a same size periodic octant of this octant throught node inode.
+ * \return Periodic octant of the same size (note: it is a stand-alone octant,
+ * may be not living in octree).
+ * The directions through the octant is periodic has to be passed as boolean flags.
+ * \param[in] inode Local index of the node target.
+ */
+Octant Octant::computeNodePeriodicOctant(uint8_t inode) const {
+    Octant degOct(this->m_dim, this->m_level, this->m_x, this->m_y, this->m_z);
+    uint32_t maxLength = uint32_t(1)<<TreeConstants::MAX_LEVEL;
+    uint32_t dh = this->getLogicalSize();
+
+    uint8_t iface1 = sm_treeConstants[m_dim].nodeFace[inode][0];
+    uint8_t iface2 = sm_treeConstants[m_dim].nodeFace[inode][1];
+    uint8_t iface3 = sm_treeConstants[m_dim].nodeFace[inode][m_dim-1];
+
+    if (!m_info[iface1] && !m_info[iface2] && !m_info[iface3]){
+        return *this;
+    }
+    else{
```

Comment:
Can 'getNodePeriodicCoord' be used to evaluate the coordinates instead of duplicating the code?

Reply:
No, they deal with periodic octant in different way.